JiraPS/Public/Find-JiraFilter.ps1

```diff
@@ -0,0 +1,89 @@
+function Find-JiraFilter {
+    # .ExternalHelp ..\JiraPS-help.xml
+    [CmdletBinding( SupportsPaging )]
+    param(
+        [Parameter(ValueFromPipeline,ValueFromPipelineByPropertyName)]
+        [string[]]$Name,
+
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [string]$AccountId,
+
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [string]$GroupName,
+
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [Object]$Project,
+
+        [Validateset('description','favourite','favouritedCount','jql','owner','searchUrl','sharePermissions','subscriptions','viewUrl')]
+        [String[]]
+        $Fields = @('description','favourite','favouritedCount','jql','owner','searchUrl','sharePermissions','subscriptions','viewUrl'),
+
+        [Validateset('description','favourite_count','is_favourite','id','name','owner')]
+        [string]$Sort,
+
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential = [System.Management.Automation.PSCredential]::Empty
+    )
+
+    begin {
+        Write-Verbose "[$($MyInvocation.MyCommand.Name)] Function started"
+
+        $server = Get-JiraConfigServer -ErrorAction Stop
+
+        $searchURi = "$server/rest/api/latest/filter/search"
+
+        [String]$Fields = $Fields -join ','
+    }
+
+    process {
+        Write-DebugMessage "[$($MyInvocation.MyCommand.Name)] PSBoundParameters: $($PSBoundParameters | Out-String)"
+        $parameter = @{
+            URI          = $searchURi
+            Method       = 'GET'
+            GetParameter = @{
+                expand = $Fields
+            }
+            Paging       = $true
+            Credential   = $Credential
+        }
+        if ($PSCmdlet.MyInvocation.BoundParameters.ContainsKey('AccountId')) {
+            $parameter['GetParameter']['accountId'] = $AccountId
+        }
+        if ($PSCmdlet.MyInvocation.BoundParameters.ContainsKey('GroupName')) {
+            $parameter['GetParameter']['groupName'] = $GroupName
+        }
+        if ($PSCmdlet.MyInvocation.BoundParameters.ContainsKey('Project')) {
+            $projectObj = Get-JiraProject -Project $Project -Credential $Credential -ErrorAction Stop
+            $parameter['GetParameter']['projectId'] = $projectObj.Id
+        }
+        if ($PSCmdlet.MyInvocation.BoundParameters.ContainsKey('Sort')) {
+            $parameter['GetParameter']['orderBy'] = $Sort
+        }
+        # Paging
+        ($PSCmdlet.PagingParameters | Get-Member -MemberType Property).Name | ForEach-Object {
+            $parameter[$_] = $PSCmdlet.PagingParameters.$_
+        }
+        if ($PSCmdlet.MyInvocation.BoundParameters.ContainsKey('Name')) {
+            foreach($_name in $Name) {
+                $parameter['GetParameter']['filterName'] = $_name
+                Write-Debug "[$($MyInvocation.MyCommand.Name)] Invoking JiraMethod with `$parameter"
+
+                $result = Invoke-JiraMethod @parameter
```

Comment:
don't assign the output of `Invoke-JiraMethod` to a variable.
This will cause the function to wait for all of `Invoke-JiraMethod` to ave finished before `$result` is ready to be used.
And thus killing the benefit of the pipeline.

Reply:
I'll change it, but either way Invoke-JiraMethod is being called once per pipeline object input and therefore I can't reproduce any delay.  If you can demonstrate otherwise, please share steps to reproduce.